Makefile

```
@@ -17,14 +17,32 @@ projectDir := $(realpath $(dir $(firstword $(MAKEFILE_LIST))))
 
 travis: make-all
 
-setup: setup-all recreate-dind-cluster
+setup: check-system-dependencies setup-all recreate-dind-cluster
 
 install: install-all
 
 clean: clean-all
 
 release: release-all
 
+check-system-dependencies:
+	@echo "== check-system-dependencies"
+ifeq (, $(shell which go))
+	@echo "golang not found in PATH"
+endif
+ifeq (, $(shell which docker))
+	@echo "docker not found in PATH"
+endif
+ifeq (, $(shell which dgoss))
+	@echo "dgoss not found in PATH"
+endif
+ifeq (, $(shell which java))
+	@echo "java not found in PATH"
+endif
+ifeq (, $(shell which kubectl))
+	@echo "kubectl not found in PATH"
+endif
+
```

Consider failing early if these commands are not found.
Otherwise these messages get lost at the top of a stream of other output.

```
which: no dgoss in (/home/richard/go/bin:/home/richard/.local/bin:/home/richard/bin:/home/richard/Downloads/google-cloud-sdk/bin:/usr/libexec/python3-sphinx:/usr/share/Modules/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/richard/.cargo/bin)
== check-system-dependencies
dgoss not found in PATH
== setup-all
make -C fake-cassandra-docker setup
make[1]: Entering directory '/home/richard/go/src/github.com/sky-uk/cassandra-operator/fake-cassandra-docker'
```